.github/workflows/documentation.yml

```diff
@@ -84,25 +66,63 @@ jobs:
           cd $GITHUB_WORKSPACE/docs
           poetry run python scripts/build_flash_parquets.py
 
+      # to be removed later. This theme doesn't support <3.9 python and our lock file contains 3.8
+      - name: install pydata-sphinx-theme
+        run: |
+          poetry run pip install pydata-sphinx-theme
+
       - name: build Sphinx docs
         run: poetry run sphinx-build -b html $GITHUB_WORKSPACE/docs $GITHUB_WORKSPACE/_build
 
-      - name: Setup Pages
-        uses: actions/configure-pages@v4
-
       - name: Upload artifact
-        uses: actions/upload-pages-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          path: '_build'
+          name: sphinx-docs
+          path: _build
 
-  # Deployment job
-  deploy:
-    environment:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
+  # this job pushes the built documentation to the docs repository
+  push:
     runs-on: ubuntu-latest
     needs: build
     steps:
-      - name: Deploy to GitHub Pages
-        id: deployment
-        uses: actions/deploy-pages@v4
+      - name: Checkout docs repo
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.repository_owner }}/docs
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: 'docs-repo'
+
+      - name: Setup SSH
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_DOCS_DEPLOY_KEY }}
+
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: sphinx-docs
+          path: sphinx-docs
+
+      - name: Determine version folder
+        id: version-folder
+        run: |
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+            echo "folder=sed/$VERSION" >> $GITHUB_OUTPUT
+          else
+            echo "folder=sed/latest" >> $GITHUB_OUTPUT
```

Comment:
This will alyways set any manual run of this workflow as the latest version. I suggest to add a case for generating docs for specific branches for testing.